src/main/java/org/truffleruby/core/rope/Rope.java

```diff
@@ -18,6 +18,9 @@
 
 public abstract class Rope {
 
+    // Useful for debugging. Setting to false allow to catch wrong usages.
```

Comment:
Interesting idea. I'd argue all usages outside of the IDE debugger support are wrong usages. But this should help find the issues at the very least.

Reply:
Ideally we'd find a way to know if we are under a Java debugger, but I don't know how to do that.